eis_toolkit/spatial_analyses/cell_based_association.py

```diff
@@ -0,0 +1,573 @@
+import os
+from numbers import Number
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import rasterio
+from beartype import beartype
+from beartype.typing import List, Optional, Tuple, Union
+from shapely import wkt
+from shapely.geometry import Polygon
+
+os.environ["USE_PYGEOS"] = "0"
+
+
+@beartype
+def cell_based_association(
+    cell_size: int,
+    geodata: List[gpd.GeoDataFrame],
+    output_path: str,
+    column: Optional[List[str]] = [""],
+    subset_target_attribute_values: Optional[List[Union[None, list]]] = [None],
+    add_name: Optional[List[Union[str, None]]] = [None],
+    add_buffer: Optional[List[Union[Number, bool]]] = [False],
+) -> gpd.GeoDataFrame:
+    """Creation of CBA matrix.
+
+    Initializes a CBA matrix from a vector file. The mesh is calculated
+    according to the geometries contained in this file and the size of cells.
+    Allows to add multiple vector data to the matrix, based on targeted shapes
+    and/or attributes.
+
+    Args:
+        cell_size: Size of the cells.
+        geodata: GeoDataFrame to create the CBA matrix. Additional
+            GeoDataFrame(s) can be imputed to add to the CBA matrix.
+        output_path: Name of the saved .tif file.
+        column: Name of the column of interest. If no attribute is specified,
+            then an artificial attribute is created representing the presence
+            or absence of the geometries of this file for each cell of the CBA
+            grid. If a target attribute is ndicated, it must be categorical.
+            Other types of attributes (numerical/string) are not managed.
+            A categorical attribute will generate as many columns (binary) in
+            the CBA matrix than values considered of interest (dummification).
+            See parameter <subset_target_attribute_values>. Additional
+            column(s) can be imputed for each added GeoDataFrame(s).
+        subset_target_attribute_values: List of values of interest of the
+            target attribute, in case a categorical target attribute has been
+            specified. Allows to filter a subset of relevant values. Additional
+            values can be imputed for each added GeoDataFrame(s).
+        add_name: Name of the column(s) to add to the matrix.
+        add_buffer: Allow the use of a buffer around shapes before the
+            intersection with CBA cells for the added GeoDataFrame(s). Minimize
+            border effects or allow increasing positive samples (i.e. cells
+            with mineralization). The size of the buffer is computed using the
+            CRS (if projected CRS in meters: value in meters).
+
+    Returns:
+        CBA matrix is created.
+    """
+
```

Comment:
You could check here that cell size is valid (positive number). This would be a logical place to check also for example that all given columns exist in the geodataframe, but I noticed you test it a bit later.

Reply:
I added a few checks at the beginning of the `cell_based_association()` function to verify input values (no empty geodataframe, no negative cell size or buffer size ...).